### ISSUE TITLE
[Bug] Deterministic Initializations (Perhaps this is Feature Request or Documentation, depends on intended behavior)

### ISSUE THREAD

**Issue by syncrostone** [cornellius-gp/gpytorch#1525]
# 🐛 Bug

The initialization of kernel parameters seems to be deterministic even when initializing multiple versions of the kernel individually.

If this is expected behavior, it would be nice to document this somewhere. In that case, this should be converted to a feature request -- ideally, I would like to be able to set it so that kernel parameters initialize randomly, and it would be nice to be able to specify what distribution these parameters are drawn from.

## To reproduce

```python
import torch
import gpytorch

a = gpytorch.kernels.RBFKernel()
b = gpytorch.kernels.RBFKernel()
print(a.lengthscale)
print(b.lengthscale)
```

** Output **
```
 tensor([[0.6931]], grad_fn=<SoftplusBackward>)
 tensor([[0.6931]], grad_fn=<SoftplusBackward>)
```

## Expected Behavior

I expected the lengthscales for the two different initializations of RBFKernel to be different.

## System information

**Please complete the following information:**
-  1.4.0
-  '1.7.1+cu101'
-  Ubuntu 16.04.6 LTS

## Additional context
I am using a version of multitask kernels (implementing GPFA in GPytorch, will post a pull request with details and implementation questions later), and symmetry in lengthscales is annoying for training at least at the start. I can initialize randomly after the fact, but ideally I would think that should be default behavior (you can always make it deterministic by setting the seed).

**Comment by Balandat** [comment]
You can add a prior on the parameters, and then use `Module.sample_from_prior` to draw a sample and load it into the module parameters. 

```
a = gpytorch.kernels.RBFKernel(lengthscale_prior=GammaPrior(1.0, 0.1))
a.sample_from_prior()
```
should work. 

Note that there may be some edge cases with the shapes though: #1317

**Comment by syncrostone** [comment]
Thanks for the quick reply! If possible it would be great to add this to the documentation so others can find it easily too.
